doc/nrf/app_approtect.rst

```diff
@@ -0,0 +1,110 @@
+.. _app_approtect:
+
+Enabling access port protection mechanism
+#########################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+.. app_approtect_info_start
+
+Several Nordic Semiconductor SoCs supported in the |NCS| offer an implementation of the access port protection mechanism (APPROTECT).
+When enabled, this mechanism blocks the debugger from read and write access to all CPU registers and memory-mapped addresses.
+Accessing these registers and addresses again requires disabling the mechanism and erasing the flash.
+In factory state, all devices come with the access port protection enabled.
+
+.. app_approtect_info_end
+
+The following table lists SoCs with information about the APPROTECT mechanism they support.
+
+.. list-table:: SoC APPROTECT matrix
+   :header-rows: 1
+
+   * - SoC
+     - Related hardware revision that supports APPROTECT
+     - Related hardware documentation
+   * - nRF52832
+     - From revision 3, build codes ``Gx0``
+     - `APPROTECT for nRF52832`_
+   * - nRF52833
+     - From revision 2, build codes ``Bx0``
+     - `APPROTECT for nRF52833`_
+   * - nRF52840
+     - From revision 3, build codes ``Fx0``
+     - `APPROTECT for nRF52840`_
+   * - nRF5340
+     - All revisions
+     - `APPROTECT for nRF5340`_
+   * - nRF9160
+     - All revisions
+     - `Debugger access protection for nRF9160`_
+
+.. _app_approtect_overview:
+
+Configuration overview
+**********************
+
+The access port protection mechanism is configured in two steps: in the hardware and in the firmware.
+
+Hardware step
+=============
+
+The hardware step consists of the verification of the mechanism state by fetching the value of ``UICR.APPROTECT``.
+This takes place at the device reset, and the mechanism state is different for different SoCs:
+
+* SoCs of the nRF52 and nRF53 Series: Default ``UICR.APPROTECT`` value prevents debugger access (APPROTECT is enabled)
+* nRF9160 SoC: Default ``UICR.APPROTECT`` value does not prevent debugger access (APPROTECT is disabled)
+
+Software step
+=============
+
+The software step takes place during startup, with the execution of a function provided by the ``hal_nordic`` module.
+This function writes a piece of debug code to the Access Port's ``APPROTECT.DISABLE`` register, which lets you debug the firmware.
+This step is valid for SoCs of the nRF52 and nRF53 Series, although the value length will differ depending on the SoC type:
+
+* SoCs of the nRF52 Series: ``0x5A`` (8-bit value)
+* SoCs of the nRF53 Series: ``0x5A0F`` (32-bit value)
```

Comment:
0x50FA50FA (double check this with datasheet)

Reply:
Section removed.